core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -157,6 +160,35 @@ abstract class AppBase(
   var mlEventLogType = ""
   var pysparkLogFlag = false
 
+  def getOrCreateExecutor(executorId: String, addTime: Long): ExecutorInfoClass = {
+    executorIdToInfo.getOrElseUpdate(executorId, {
+      new ExecutorInfoClass(executorId, addTime)
+    })
+  }
+
+  /**
+   * Retrieves cluster information based on executor nodes.
+   * If executor nodes exist, calculates the number of hosts and total cores,
+   * and extracts executor and driver instance types (databricks only)
+   *
+   * @return
+   */
+  def getClusterInfo: Option[ClusterInfo] = {
+    // Extracts instance types from properties (databricks only)
+    val executorInstance = sparkProperties.get("spark.databricks.workerNodeTypeId")
```

Comment:
nit: something we don't need it now but down the road we may want to have a helper depending on the platform. so we can extend and customize code easily. but it is fine to do that later.